mu-plugins/central-hub/tests/phpunit/integration/meta-data/saveCustomFields.php

```diff
@@ -0,0 +1,195 @@
+		$this->assertSame( [], $_POST['events'] );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'event-date', true ) );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'event-time', true ) );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'venue-name', true ) );
+	}
+
+	/**
+	 * Test save_custom_fields() should delete actual post meta values set in $_POST when they equal the config delete
+	 * state.
+	 */
+	public function test_should_delete_actual_post_meta_values_set_in_POST_when_they_equal_the_config_delete_state() {
+		// Custom fields config.
+		$config = [
+			'event-date' => [
+				'is_single'    => true,
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+			'event-time' => [
+				'is_single'    => true,
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+			'venue-name' => [
+				'is_single'    => true,
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+		];
+		// Add post meta to the database.
+		add_post_meta( $this->post->ID, 'event-date', '' );
+		add_post_meta( $this->post->ID, 'event-time', '' );
+		add_post_meta( $this->post->ID, 'venue-name', '' );
+		// $_POST contains the actual value for each custom field.
+		$_POST = [
+			'events' => [
+				'event-date' => '',
+				'event-time' => '',
+				'venue-name' => '',
+			],
+		];
+
+		save_custom_fields( $config, 'events', $this->post->ID );
+
+		// Check that post meta keys are deleted when their value equals the config delete state.
+		$this->assertSame( '', $_POST['events']['event-date'] );
+		$this->assertSame( '', $_POST['events']['event-time'] );
+		$this->assertSame( '', $_POST['events']['venue-name'] );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'event-date', true ) );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'event-time', true ) );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'venue-name', true ) );
+	}
+
+	/**
+	 * Test save_custom_fields() saves the sanitized current post meta from the custom field into the database.
+	 */
+	public function test_should_save_the_sanitized_current_post_meta_from_the_custom_field_into_the_database() {
+		// Custom fields config.
+		$config = [
+			'event-date' => [
+				'is_single'    => true,
+				'default'      => '',
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+			'event-time' => [
+				'is_single'    => true,
+				'default'      => '',
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+			'venue-name' => [
+				'is_single'    => true,
+				'default'      => '',
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+		];
+		// Add post meta to the database.
+		add_post_meta( $this->post->ID, 'event-date', '' );
+		add_post_meta( $this->post->ID, 'event-time', '' );
+		add_post_meta( $this->post->ID, 'venue-name', '' );
```

Comment:
@rgadon107 Let's not store the values in the database before this test. Why? Less code to manage.

Reply:
Removed.